### PR TITLE
sql/schemachanger: extend DML injection tests

### PR DESCRIPTION
**Pull request by fqazi**
These changes address the following things in the declarative schema changer
1. Extend DML injection testing into more schema changes in our supported set of changes
2. Update rollback testing to include DML for both happy/unhappy paths (this allows for more coverage).
3. Address the constraint name not being properly populated in certain error cases while a schema change is active with DML.

### CODE REVIEW

File: pkg/sql/schemachanger/scplan/internal/rules/current/dep_drop_constraint.go [hunk 1]
```diff
@@ -61,10 +61,41 @@ func init() {
...
+	// Constraint name should be cleared right before the constraint is no
+	// longer visible.
+	registerDepRuleForDrop(
+		"Constraint should be hidden before name",
+		scgraph.SameStagePrecedence,
+		"constraint-name", "constraint",
+		scpb.Status_ABSENT, scpb.Status_ABSENT,
+		func(from, to NodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.Type((*scpb.ConstraintWithoutIndexName)(nil)),
```
Comment (chengxiong-ruan) [inline pkg/sql/schemachanger/scplan/internal/rules/current/dep_drop_constraint.go:80]: use `isConstraintWithIndexName` ?

File: pkg/sql/schemachanger/scplan/internal/rules/current/dep_add_constraint.go [hunk 2]
```diff
@@ -47,4 +47,20 @@ func init() {
...
+	// Constraint name should be assigned right before it becomes visible, otherwise
+	// we won't have the correct message inside errors.
+	registerDepRule(
+		"simple constraint visible before name",
+		scgraph.SameStagePrecedence,
+		"simple-constraint", "constraint-name",
+		func(from, to NodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.TypeFilter(rulesVersionKey, isNonIndexBackedConstraint),
+				to.Type((*scpb.ConstraintWithoutIndexName)(nil)),
```
Comment (chengxiong-ruan) [inline pkg/sql/schemachanger/scplan/internal/rules/current/dep_add_constraint.go:60]: use `isConstraintWithIndexName` ?

**Review by chengxiong-ruan — APPROVED** [review]
nice~

**Comment by fqazi** [comment]
@chengxiong-ruan TFTR!

**Pull request merged**